docs/usage.md

```diff
@@ -69,119 +115,360 @@ cat ./compilation.json
+
+Make sure, that you use the same analysis configuration when updating a run,
+because a changed analysis configuration can make new reports to appear or
+reports to disappear.
+
+### Programmer checking new bugs in the code after local edit (and compare it to a central database)<a name="compare"></a>
+Say that you made some local changes in your code (automatically fixing
+example program) and you wonder whether you fixed any bugs. Each bug has a
+unique hash identifier that is independent of the line number, therefore
+resistant to shifts in the source code. This way, newly introduced bugs can be
+detected, compared to a central CodeChecker report database.
+
+If you stored [first analization of unmodified](#run-the-analysis), example
+project and made [automatic fixing](#automatic-fixing) then you can compare the
+result between stored and locally analyzed example project.
 
-### Analysis Failures <a name="analysis-failures"></a>
+1. [Analyze unmodified example project](#step-2-analyze-your-code)
+```sh
+CodeChecker analyze ./compile_commands.json --output ./reports --enable-all
+```
+
+2. [Store the result on your local database](#step-5-store-analysis-results-in-a-codechecker-db-and-visualize-results)
+```sh
+CodeChecker store ./reports --name example --url http://localhost:8555/Default
+```
+
+3. Do [automatic fix](#automatic-fixing)
+```sh
+CodeChecker fixit --checker-name modernize-deprecated-headers --apply \
+    ./reports
+```
+
+4. [Re-analyze your code](#using-incremental-build-on-modified-files). You are
+well advised to use the same `analyze` options as you did in the first
+analization session: the same checkers enabled, the same analyzer options, etc.
+```sh
+CodeChecker analyze ./compile_commands.json --output ./reports --enable-all
+```
+
+5. Compare your local analysis to the central one
+```sh
+CodeChecker cmd diff --basename example --newname ./reports --resolved \
+    --url http://localhost:8555/Default
+```
+
+### Using diff command on the local filesystem<a name="using-diff"></a>
+Developer can compare two results of analyses without upload them to a
+central server. In this case the analysis result should be stored in different
+result directory.
+```sh
+CodeChecker cmd diff --basename ./reports --newname ./reports-ctu --new
+```
 
-The `reports/failed` folder contains all build-actions that
-were failed to analyze. For these there will be no results.
+ You can also use JSON format output of `CodeChecker cmd diff` command if
+ you want to use it for further processing by an other program.
+
+## Step 7: Fine tune Analysis configuration <a name="step-6"></a>
+### Analysis Failures <a name="step-7"></a>
+The `reports/failed` folder contains all build-actions that were failed to
+analyze. For these there will be no results.
```

Comment:
It is not true, because Clang-Tidy will generate reports for compiler failures.

Reply:
Fixed.